diff --git a/www/saml2/idp/metadata.php b/www/saml2/idp/metadata.php
index 0b5f943..3c2ca10 100644
--- a/www/saml2/idp/metadata.php
+++ b/www/saml2/idp/metadata.php
@@ -67,6 +67,15 @@ try {
 	$ssol = $metadata->getGenerated('SingleSignOnService', 'saml20-idp-hosted');
 	$slol = $metadata->getGenerated('SingleLogoutService', 'saml20-idp-hosted');
 
+	// Adding a POST binding forces UAA to generate an auto-POST-ing page instead of doing
+	// a simple 302 redirect. This allows us to ensure embedded Javascript is working, e.g.
+	// if there's a change to Content-Security-Policy headers that would prevent the inline
+	// javascript on that page.
+	$metaArray['SingleSignOnService'][] = array(
+		'Binding' => 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
+		'Location' => $ssol,
+	);
+
 	if (is_array($ssob)) {
 		foreach ($ssob as $binding) {
 			$metaArray['SingleSignOnService'][] = array(
